lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinitionBuilder.cs

```diff
@@ -28,19 +32,21 @@ namespace Org.Apache.REEF.IMRU.API
     /// <seealso cref="IMRUJobDefinition" />
     public sealed class IMRUJobDefinitionBuilder
     {
-        private IConfiguration _configuration;
+        private static readonly Logger Logger = Logger.GetLogger(typeof(IMRUJobDefinitionBuilder));
+
         private string _jobName;
 
+        private static readonly IConfiguration EmptyConfiguration =
+            TangFactory.GetTang().NewConfigurationBuilder().Build();
+
         /// <summary>
-        /// Set the Configuration used to instantiate the IMapFunction, IReduceFunction, IUpdateFunction and all codec instances
+        /// Constructor
         /// </summary>
-        /// <param name="configuration">The Configuration used to instantiate the IMapFunction instance.</param>
-        /// <seealso cref="IMRUConfiguration{TMapInput,TMapOutput,TResult}" />
-        /// <returns>this</returns>
-        public IMRUJobDefinitionBuilder SetConfiguration(IConfiguration configuration)
+        public IMRUJobDefinitionBuilder()
         {
-            _configuration = configuration;
-            return this;
+            MapInputPipelineDataConverterConfiguration = EmptyConfiguration;
```

Comment:
That will not happen since if user wants to give something else he will
create new instance. Note that there is no get method. So it cannot be
appended.

Dhruv
On Aug 6, 2015 16:22, "Julia Wang" <notifications@github.com> wrote:

> In lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinitionBuilder.cs
> <https://github.com/apache/incubator-reef/pull/334#discussion_r36477825>:
>
> >          {
> > -            _configuration = configuration;
> > -            return this;
> > +            MapInputPipelineDataConverterConfiguration = EmptyConfiguration;
>
> If the purpose is to merge it with others, then sharing the same empty
> instance is OK. If you might add some other settings in the configuration
> respectively, then sharing may cause problem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/334/files#r36477825>.
>